### ISSUE TITLE
Compendiums for weapons and gear

### ISSUE THREAD

**Issue by spidercourgan** [OctarineSourcerer/cyberpunk2020#55]
Very good work. Thank you for this.
It is possible to get the compendiums for weapons and gear?
I saw it in the screenshots.
Maybe  from user Drowzero0??
Thank you for an answer

**Comment by drowzero0** [comment]
i have currently implemented around 30 weapons with every settings and around 200 other items, but it is far from completion.
if you want, i can complete those compendiums and send em over when i am done.
its just some .DB files and loads of pictures which i thought would be most fitting.

**Comment by spidercourgan** [comment]
that would be very nice..
we played on cyberpunk red system and there I implanted the hole core rulebook. (weapons, gear, cybersystems)
But I can't migrate it to cyberpunk 2020.

**Comment by OctarineSourcerer** [comment]
I think drowzero0 made their own sets of items :) 

This is on the roadmap though - there was someone on the discord who is currently working on something like a csv for that, but if you do end up completing the compendiums, feel free to send em over and I'll see what I can do.

I _will_ say that whatever ends up in the system won't have art from the books, or descriptions from the books, in accordance with R Talsorian's homebrew content policy under virtual tabletop systems https://rtalsoriangames.com/homebrew-content-policy/
The same will be true for any non-book art - I'd only want to use any art I actually have some level of license to use.

**Comment by drowzero0** [comment]
after reading the whole policy from your link it is kinda problematic to use those descriptions and images, as i am using them in my private session.
but i read somewhere, that it is possible to get a using-license if you ask them. they just want to make sure the stuff they created is not used without them knowing.
as example i am using this french site (i do not speak french but stats are clear enough) to get all my weapons into foundry and also will add all other weapons listed there: http://cyberpunk.asia/ which i am using as a reference for playing for years now
i will delete all textdescriptions i used so far. before handing those .db files to you.
also created misc files will be deleted that i only use for my storyboard.

but what about this site?
https://cyberpunk.fandom.com/wiki/Main_Page 
they are using kind of every text from the books, and also a load of pictures from there. so i guess it is possible to use talsorians content. i will get in touch with them if you want that @OctarineSourcerer

**Comment by drowzero0** [comment]
just upgraded and just want to let @spidercourgan know:
![image](https://user-images.githubusercontent.com/62395017/115159851-7fd26c00-a095-11eb-9ad4-2df951e37df6.png)
but there are around 1000s items more. so i would guess i will take around 1 month to get Everything from core and chromebook(s) into the database.

**Comment by spidercourgan** [comment]
OK, that is much work. The weapons, I put in a few weeks ago are shown and they are working, but I can't edit them anymore.

**Comment by OctarineSourcerer** [comment]
Yes, it is a lot of work I'm afraid, which is a good part of why it's not done yet - my priority so far has been to get the system from 0 to something usable first.

@drowzero0 I appreciate that. With the policy blessing at least use of stat blocks and rules emulation, I don't see why I shouldn't email them - I'll try and do that in the next few days. I don't think they'll budge, but I feel it's reasonable to ask.

**Comment by drowzero0** [comment]
i created my first shareable itemlist.
it contains 338 Pistols, from light to very heavy.
![image](https://user-images.githubusercontent.com/62395017/118357065-747d3e00-b578-11eb-8bf0-1ddc76cf2f6a.png)

i also have images for every weapon (on my personal database only), but they are not linked, as @OctarineSourcerer mentioned.

i am not sure which files you need exactly, i have the items.db and folders.db, but i am unable to upload them here.

**Comment by OctarineSourcerer** [comment]
> i created my first shareable itemlist.
> it contains 338 Pistols, from light to very heavy.
> 
> i also have images for every weapon (on my personal database only), but they are not linked, as @OctarineSourcerer mentioned.
> 
> i am not sure which files you need exactly, i have the items.db and folders.db, but i am unable to upload them here.
> 
> i also have the compendium file pistols.db.

woah that's awesome! Are you able to email those files to me, or upload them somewhere and link them? I can see what I can do :)

**Comment by drowzero0** [comment]
i sent you the db files yesterday by email, not sure if you received it?

**Comment by OctarineSourcerer** [comment]
> i sent you the db files yesterday by email, not sure if you received it?

Hey! Sorry, had a super busy couple days. Your pistol compendium should be in `develop` now though! :) Added bonus, it got me to write code for migrating entities in compendiums too :P 

I might see about putting the template file you used somewhere so it's easy for anyone to make a compendium like that, though I'm not sure where.

**Comment by drowzero0** [comment]
i guess i cant test your actuall version without breaking my current world?
or do you have an idea how to?

for the future:
i am currently creating the MISC and the Armour template.json too, i want to get a few more categories for weapons ,like rifles and heavy weapons, too.
so give me some time, when i am done i will mail you the db files.

**Comment by OctarineSourcerer** [comment]
> i guess i cant test your actuall version without breaking my current world?
> or do you have an idea how to?
> 
> for the future:
> i am currently creating the MISC and the Armour template.json too, i want to get a few more categories for weapons ,like rifles and heavy weapons, too.
> so give me some time, when i am done i will mail you the db files.

I don't think it'd break, but just in case, make a copy of your current world, and test on that? :) or a new world and test, either works.

Awesome, that sounds great, thank you.

**Comment by cybersmily** [comment]
You need weapon data? I got a json file with the weapon stats over on my site, including source and page, https://cybersmily.net. Doesn't have the descriptions for them. Also have skills, roles, and cyberware as well.

**Comment by drowzero0** [comment]
@cybersmily if its ok for you, i would compare the 416 weapons from your site and those 1000+ weapons i gathered and add the missing ones to get a whole complete weapon compendium.

**Comment by cybersmily** [comment]
@drowzero0 I didn't know what format you have the data in. Figured to help ease the workload with something already formatted ;) Thanks for the compliment, I do my best. Also if you want to lift some of the code base, I have a github repo.

**Comment by OctarineSourcerer** [comment]
Sweey lordy that's a lot of data @cybersmily. Really solid site in general, nicely done :)

**Comment by cybersmily** [comment]
@OctarineSourcerer thanks for the compliment. If you need the json files let me know

**Comment by cybersmily** [comment]
@drowzero0 one thing about the cyberpunk.asia site just so you know, the owner lifted most of the site's content from other fan sites without permission. I talk with Wisdom Datafortress2020, and though Wisdom likes that his stuff was translated to French, he doesn't appreciate it was done without asking. With the homebrew policy you need to get permissions for the source owners just a FYI.

**Comment by drowzero0** [comment]
> @drowzero0 one thing about the cyberpunk.asia site just so you know, the owner lifted most of the site's content from other fan sites without permission. I talk with Wisdom Datafortress2020, and though Wisdom likes that his stuff was translated to French, he doesn't appreciate it was done without asking. With the homebrew policy you need to get permissions for the source owners just a FYI.

i am not sure, what to do now.
do i have to ask wisdom000 from Datafortress, and all other used websites at cyberpunk.asia for their permissions, to use the named weapons, vehicles and so on?
did you talk to the owner of datafortress2020 already? 

as i mentioned in another issue i do not have a good knowledge about licensing and stuff like that.
if there are any hints for me, i would appreciate that to make less errors in the future. 
if i made any wrongdoings then i did without bad intentions and because of my lack in knowledge in this segment.

**Comment by OctarineSourcerer** [comment]
If that's the case, it might be best to go forward with @cybersmily's data for the sake of keeping this all as okay-with-everyone as possible. If I understand it rightly, they've asked Wisdom for the data, and so we can use it?

No worries @drowzero0, I don't think we're actually in any trouble here, nor is there malicious intent. Sometimes things just slip by :) If we could use cybersmily's JSONs (and if needed for your workflow @drowzero0 convert them to CSV with something like http://convertcsv.com/json-to-csv.htm), then I think that'd work.

**Comment by cybersmily** [comment]
licensing is crazy. I try to keep with cannon stuff from my site just for that purpose. As for Wisdom and datafortress2020, I can ask him. I don't know about the other stuff on asia cyberpunk stuff and where all of it. I'm sure if you each out to asia site owner ( I think they have an email link on the site), you can ask them as well.

**Comment by cybersmily** [comment]
Oh, if you want to add in images @drowzero0 each out to J Gray over on R Talsorian. He's the guy to get permission for use of images from the books.

**Comment by cybersmily** [comment]
One other thing, I got the converter for the cyberware json I have, does some property mapping etc. Would like to know would it be preferable to have all in one compendium or split out into bioware, neuralware, implants, etc. each with their own .db file?

**Comment by drowzero0** [comment]
> Oh, if you want to add in images @drowzero0 each out to J Gray over on R Talsorian. He's the guy to get permission for use of images from the books.

i will get in contact with him. i will also explain what we are doing here, maybe we get a permission to use those pictures in those .dbs.

i will also get in contact with the owner of cyberpunk.asia and all mentioned sources he uses.
it would be great if you can ask Wisdom. 

i will check all used items so far, and if they are from a diffrent souce i will clearout the usage-rights.
hopefully i get something i can show you afterwards like an official statement from each that can be pinned into the db or as a license note into the repo.

**Comment by drowzero0** [comment]
> One other thing, I got the converter for the cyberware json I have, does some property mapping etc. Would like to know would it be preferable to have all in one compendium or split out into bioware, neuralware, implants, etc. each with their own .db file?

i am not sure, i am currently testing out how to go on with misc items:
my solution looks like this, so far:
![image](https://user-images.githubusercontent.com/62395017/119346746-e7826500-bc9a-11eb-8317-499ab4eb85cd.png)

so its more a differing on "flavor" than on the db itself.
but if there are a few hundred cyberware parts from diffrent sources guess the categories as you mentioned would be easier to sort.

**Comment by cybersmily** [comment]
do we want to have a separate field for source or add it to flavor? If we add to flavor, I just need to update my converter to insert it there.

**Comment by drowzero0** [comment]
if there is an extra field for all items, then it would look more tidied up.
but that decision is for @OctarineSourcerer to be made.

**Comment by cybersmily** [comment]
Will be adding one with my next push. Figure it can be displayed right under flavor.

**Comment by drowzero0** [comment]
> licensing is crazy. I try to keep with cannon stuff from my site just for that purpose. As for Wisdom and datafortress2020, I can ask him. I don't know about the other stuff on asia cyberpunk stuff and where all of it. I'm sure if you each out to asia site owner ( I think they have an email link on the site), you can ask them as well.

i just wrote an email to R.Talsorian Games, Datafortress -Wisdom and the owner of cyberpunk.asia, for permissions.
as soon as i have any news i will let you guys know.

**Comment by cybersmily** [comment]
talked with Wisdom today, he's cool with using his stuff as long as well credit to his site.

**Comment by drowzero0** [comment]
do i have to change the source to the websitelink then?

**Comment by OctarineSourcerer** [comment]
> if there is an extra field for all items, then it would look more tidied up.
> but that decision is for @OctarineSourcerer to be made.

Oop sorry for the late reply here, I think putting it under flavour works just fine here :) 

Cheers for talkin' to peeps, wisdom being good with it is awesome news!

**Comment by OctarineSourcerer** [comment]
> do i have to change the source to the websitelink then?

I don't think so - that data is originally from the sourcebooks. I think wisdom being cool with us using the data is at the "import what wisdom has gathered" level, though ofc I'm more than happy to credit wisdom in the module readme itself

**Comment by cybersmily** [comment]
hey, instead of using this issue as a forum, is there interest in creating a discord to talk about design, task ownership, strategy etc?

**Comment by downshiftdata** [comment]
FYI, #124 is awaiting review. This adds an SMG compendium.
